<a href="https://colab.research.google.com/github/dacdemon/ITD/blob/main/INTEGRADOR%20CORIA%20DAMIAN.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Práctica 4: Miremos un poco a nuestro planeta.
##Alumno: Damián Coria
##Comisión: 2

El Ministerio de Energía busca reducir progresivamente la dependencia de los combustibles fósiles y aumentar la participación de fuentes renovables en la matriz energética.

Ante esta situación, se solicita realizar un análisis de datos históricos de distintos países con el objetivo de estudiar la evolución de las principales fuentes de energía renovable, como la energía hidroeléctrica, solar y eólica.

El propósito del análisis será identificar tendencias, comparar el crecimiento de estas tecnologías y evaluar cuál de ellas presenta mejores condiciones para acompañar una transición hacia una matriz energética con menor dependencia de combustibles fósiles.

A partir de los resultados obtenidos, se elaborará una conclusión basada en los datos analizados.

In [ ]:
# Importamos las librerias necesarias para esta praactica (pandas, matplot y seaborn).

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Cargamos los archivos csv desde un repositorio web.

url = "https://raw.githubusercontent.com/dacdemon/ITD/refs/heads/main/renewable_energy_by_country.csv"
df = pd.read_csv(url)


# El DataFrame original es muy grande, por lo que se realiza una exploración inicial para conocer sus dimensiones, las columnas disponibles y el período de los datos. A partir de este análisis se seleccionarán las variables y registros más relevantes, respetando los requisitos mínimos de la consigna: 3000 filas y 5 columnas.

In [ ]:
# Se crea una función de análisis para reutilizarla y verificar los cambios realizados en el DataFrame.

def analizar_dataset(df):

    # Dimensiones:

    print("\nDimensiones:")
    dimensiones = df.shape
    print(dimensiones[0], "Filas")
    print(dimensiones[1], "Columnas")

    # columnas disponibles:

    print("\nColumnas:")
    print(df.columns)

    # se analiza dos columnas claves: paises y año

    print("\nCantidad de países:", df["country"].nunique())
    print("Año inicial:", df["year"].min())
    print("Año final:", df["year"].max())

analizar_dataset(df)


Dimensiones:
17265 Filas
29 Columnas

Columnas:
Index(['country', 'year', 'iso_code', 'population', 'gdp',
       'primary_energy_consumption', 'energy_per_capita',
       'fossil_share_energy', 'fossil_fuel_consumption',
       'electricity_generation', 'renewables_share_energy',
       'renewables_share_elec', 'renewables_consumption',
       'renewables_electricity', 'hydro_electricity', 'hydro_share_elec',
       'hydro_share_energy', 'solar_electricity', 'solar_share_elec',
       'solar_share_energy', 'wind_electricity', 'wind_share_elec',
       'wind_share_energy', 'other_renewable_electricity',
       'other_renewables_share_elec', 'other_renewables_share_energy',
       'renewables_electricity.1', 'renewables_share_elec.1',
       'renewables_share_energy.1'],
      dtype='object')

Cantidad de países: 220
Año inicial: 1900
Año final: 2025


# Se seleccionaron estas columnas porque están directamente relacionadas con el objetivo del análisis. Permiten comparar la participación de los combustibles fósiles con distintas fuentes de energía renovable, analizar su evolución a lo largo del tiempo y estudiar las relaciones entre las variables.

In [ ]:
# columnas elegidas.

variables = [
    "country",
    "year",
    "fossil_share_energy",
    "hydro_share_energy",
    "solar_share_energy",
    "wind_share_energy",
    "other_renewables_share_energy",
    "primary_energy_consumption"]

df_analisis = df[variables]

print(df_analisis.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17265 entries, 0 to 17264
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   country                        17265 non-null  object 
 1   year                           17265 non-null  int64  
 2   fossil_share_energy            4459 non-null   float64
 3   hydro_share_energy             4459 non-null   float64
 4   solar_share_energy             4459 non-null   float64
 5   wind_share_energy              4459 non-null   float64
 6   other_renewables_share_energy  4459 non-null   float64
 7   primary_energy_consumption     10355 non-null  float64
dtypes: float64(6), int64(1), object(1)
memory usage: 1.1+ MB
None


# En el análisis anterior se observó una gran cantidad de valores nulos.
#Como el dataset comienza en 1900, se plantea la hipótesis de que estos valores pueden deberse a la falta de registros de algunas tecnologías energéticas en los años más antiguos.
# Por este motivo, se analizará desde qué año existe información completa para las variables seleccionadas.

In [ ]:
df_analisis = df_analisis.dropna()

print("Dimensiones:", df_analisis.shape)
print("Cantidad de países:", df_analisis["country"].nunique())
print("Período:", df_analisis["year"].min(), "-", df_analisis["year"].max())

Dimensiones: (4459, 8)
Cantidad de países: 79
Período: 1965 - 2024


# La hipótesis planteada anteriormente era correcta: los datos completos para las variables seleccionadas se concentran en años más recientes, desde 1965 hasta 2024.
# De esta manera, se obtiene un DataFrame de 4459 filas y 8 columnas, que cumple con los requisitos mínimos de la consigna y mantiene datos relevantes para el análisis.

In [ ]:
# Procedemos a reducir el DataFrame original y lo reemplazamos por el conjunto de datos elejido

columnas_seleccionadas = [
    "country",
    "year",
    "fossil_share_energy",
    "hydro_share_energy",
    "solar_share_energy",
    "wind_share_energy",
    "other_renewables_share_energy",
    "primary_energy_consumption"]

# Se conservan las columnas seleccionadas y se eliminan las filas con valores nulos

df = df[columnas_seleccionadas].dropna()

# Utilizamos la funcion previamente creada para analizar como quedo nuestro dataframe

analizar_dataset(df)


Dimensiones:
4459 Filas
8 Columnas

Columnas:
Index(['country', 'year', 'fossil_share_energy', 'hydro_share_energy',
       'solar_share_energy', 'wind_share_energy',
       'other_renewables_share_energy', 'primary_energy_consumption'],
      dtype='object')

Cantidad de países: 79
Año inicial: 1965
Año final: 2024


# Una vez preparado el dataset y seleccionados los datos relevantes para nuestro análisis, se inicia el desarrollo del Trabajo Práctico Integrador.